slice/partition.go

```diff
@@ -0,0 +1,30 @@
+package slice
+
+// Partition creates partitions of a standard maximum size.
+func Partition[T any](s []T, partitionSize int) [][]T {
+	if len(s) == 0 || partitionSize <= 0 {
+		return [][]T{}
+	}
+
+	partitions := make([][]T, 0, len(s)/partitionSize)
```

Comment:
the capacity should be rounded up `make([][]T, 0, (len(s) + partitionSize - 1)/partitionSize)`